### PR TITLE
sys/tiny_strerror: add missing error codes

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

When double-checking the error codes provided by newlib by default (without magic defines, such as `__LINUX_ERRNO_EXTENSIONS__` or `__CYGWIN__`), some where still missing in `tiny_strerror()`. This adds the missing ones.

This in turn showed that three errno codes were missing in the avr-libc compat `errno.h`, which are added as well.

### Testing procedure

Murdock should double check that the added errno codes indeed are defined by default.

### Issues/PRs references

None

### CODE REVIEW

File: sys/tiny_strerror/tiny_strerror.c [hunk 1]
```diff
@@ -43,10 +43,14 @@ static const char *lookup[] = {
     [EDEADLK]           = "-EDEADLK",
     [EDESTADDRREQ]      = "-EDESTADDRREQ",
     [EDOM]              = "-EDOM",
+    [EDQUOT]            = "-EDQUOT",
     [EEXIST]            = "-EEXIST",
     [EFAULT]            = "-EFAULT",
     [EFBIG]             = "-EFBIG",
+    [EFTYPE]            = "-EFTYPE",
```
Comment (maribu) [inline sys/tiny_strerror/tiny_strerror.c:50]: This is not present in `man errno` :/
```suggestion
```

**Review by kaspar030 — APPROVED** [review]
ACK.

bors merge

**Comment by maribu** [comment]
bors retry

**Comment by miri64** [comment]
bors merge

**Comment by maribu** [comment]
bors retry

**Comment by maribu** [comment]
bors retry

**Comment by maribu** [comment]
bors cancel

**Comment by maribu** [comment]
bors merge

**Comment by maribu** [comment]
bors cancel
bors merge

**Comment by kaspar030** [comment]
huh

**Comment by kaspar030** [comment]
the label check is missing

**Review by benpicco — APPROVED** [review]

**Comment by kaspar030** [comment]
bors merge

**Review by nandojve — APPROVED** [review]

**Comment by maribu** [comment]
bors merge

I ran `dist/tools/insufficient_memory/create_makefile.ci.sh` for each and every app that contained `strerror`. Let's see if that helps

**Comment by benpicco** [comment]
bors merge

**Comment by maribu** [comment]
bors cancel
bors merge

**Comment by maribu** [comment]
bors merge

**Comment by benpicco** [comment]
I don't like that tiny_strerror got so large… but maybe we can trim it down a bit as a follow-up

bors merge

**Comment by benpicco** [comment]
bors merge

**Pull request merged**